File: appengine/env.ts
export type Env = "prod" | "dev"

export function env(): Env {
	if (process.env.NODE_ENV === "production") {
		return "prod"
	}
	return "dev"
}

export const devPort = 8080

export function devBaseURL(): string {
	return `http://localhost:${8080}`
}

File: appengine-go/env.go
package main

import (
	"fmt"
	"os"
)

type Env string

const (
	Dev  Env = "dev"
	Prod Env = "prod"
)

func env() Env {
	_, ok := os.LookupEnv("GAE_DEPLOYMENT_ID")
	if ok {
		return Prod
	}
	return Dev
}

func isDev() bool {
	return env() == Dev
}

const devPort = "8080"

func devAddr() string {
	return fmt.Sprintf("localhost:%s", devPort)
}

func devBaseURL() string {
	return "http://" + devAddr()
}
